homeassistant/components/abode/switch.py

```diff
@@ -52,15 +51,33 @@ def is_on(self):
 class AbodeAutomationSwitch(AbodeAutomation, SwitchDevice):
     """A switch implementation for Abode automations."""
 
+    async def async_added_to_hass(self):
+        """Subscribe Abode events."""
+        await super().async_added_to_hass()
+
+        signal = SIGNAL_TRIGGER_AUTOMATION.format(self.entity_id)
+        async_dispatcher_connect(self.hass, signal, self.trigger)
+
     def turn_on(self, **kwargs):
-        """Turn on the device."""
-        self._automation.set_active(True)
+        """Enable the automation."""
+        if self._automation.enable(True):
+            self.schedule_update_ha_state()
```

Comment:
This is required based on there being no web socket events for automations being disabled or enabled (unlike normal devices in Abode). How the abode integration is currently setup, when a device is switched on/off, it calls a method in abodepy that will execute the command and also immediately refresh the device's JSON data. However, the abode integration waits until it receives a web socket event of the device state changing to execute a call back which calls `self.schedule_update_ha_state()`. Again, this doesn't happen with automations hence the reason I'm calling `self.schedule_update_ha_state()` here.